docs/02-architecture/06-client-side-architecture.md

```diff
@@ -0,0 +1,285 @@
+
+`<link rel="prefetch" href="@Static("javascripts/app.js")">`
+
+#### Cloudwatch beacon
+
+This is where we would ping cloudwatch.. if we had anything to ping about..
+
+### Inline non-blocking JS
+
+#### getUserData
+
+The [getUserData](https://github.com/guardian/frontend/blob/master/common/app/templates/inlineJS/nonBlocking/getUserData.scala.js) util puts the current logged-in user's data in the JS config ready to be used by other JS.
+
+#### detectAdblock
+
+Secrets.
+
+#### showUserName
+
+[Puts the user's username](https://github.com/guardian/frontend/blob/master/common/app/templates/inlineJS/nonBlocking/showUserName.scala.js) into the header (increases perceived rendering speed as their username is there before the app.js has to download and parse).
+
+#### Ophan config
+
+Gets the [Ophan browserId](https://github.com/guardian/frontend/blob/master/common/app/templates/inlineJS/nonBlocking/ophanConfig.scala.js) which is used across analytics to tie data together.
+
+### Bootstraps
+
+In javascripts/bootstraps we define all the entry points for each bundle described in [requirejs.js](). 
+
+The top level entry points which call the bootstrap initialisation of all other bundles are [enhanced/main.js](), [standard/main.js](), [admin.js]() (for frontend.gutools, not theguardian.com), [commercial.js]() and [video-embed.js]() (initialised when there is a video embed from [videoEmbed.scala.html]()).
+
+### app.js
+
+- [Builds a bundle](https://github.com/guardian/frontend/blob/master/grunt-configs/requirejs.js#L47) for [standard/main bootstrap](https://github.com/guardian/frontend/blob/master/static/src/javascripts/bootstraps/standard/main.js)
+	- Includes the [boot.js](https://github.com/guardian/frontend/blob/master/static/src/javascripts/boot.js) in the bundle
+	
+```js
+boot: {
+	options: {
+		name: 'boot',
+		out: options.staticTargetDir + 'javascripts/boot.js',
+		include: 'bootstraps/standard/main',
+		insertRequire: ['boot'],
+		exclude: [
+			'text',
+			'inlineSvg'
+		]
+	}
+}
+```
+
+- And [concatenates that](https://github.com/guardian/frontend/blob/master/grunt-configs/concat.js#L8) with the [curl-domReady.js](https://github.com/cujojs/curl) (Curl module loader that waits for domReady)
+
+```js
+app: {
+	src: [
+		options.staticSrcDir + 'javascripts/components/curl/curl-domReady.js',
+		options.staticTargetDir + 'javascripts/boot.js'
+	],
+	dest: options.staticTargetDir + 'javascripts/app.js'
+}
+```
+
+
+#### Standard main.js
+
+To quote the file:
+
+> // This file is intended to be downloaded and run ASAP on all pages by all readers.
+
+> // While it's ok to run code from here that requires specific host capabilities, it should manage failing gracefully by itself.
+
+>// Assume *nothing* about the host...
+
+> // This also means you should think *very hard* before adding modules to it, in particular 3rd party modules.
+
+> // For this file, performance and breadth of support should take priority over *anything*…
```

Comment:
Style thing: I don't think the above needs to be presented as comments, it seems unnecessary